api/src/main/java/jakarta/data/page/Page.java

```diff
@@ -17,39 +17,179 @@
+     * page was obtained.
+     *
+     * @return the request for the current page; will never be {@code null}.
+     */
+    PageRequest<T> pageRequest();
+
+    /**
+     * <p>Returns the {@linkplain PageRequest page request} for which this page
+     * was obtained.</p>
+     *
+     * <p>This method is provided for when {@linkplain Query query language} is used to
+     * return a result of different type than the entity that is being queried.
+     * This method allows the {@link PageRequest} to be returned for the
+     * type of entity class that was queried.</p>
+     *
+     * @param <E>         entity class of the attributes that are used as sort criteria.
+     * @param entityClass entity class of the attributes that are used as sort criteria.
+     * @return the request for the current page; will never be {@code null}.
+     */
+    <E> PageRequest<E> pageRequest(Class<E> entityClass);
+
+    /**
+     * Returns a request for the {@linkplain PageRequest#next() next} page, or <code>null</code> if it is known that there is no next page.
+     *
+     * @return a request for the next page.
+     */
+    PageRequest<T> nextPageRequest();
+
+    /**
+     * <p>Returns a request for the {@linkplain PageRequest#next() next} page,
+     * or <code>null</code> if it is known that there is no next page.</p>
+     *
+     * <p>This method is useful when {@linkplain Query query language} is used to
+     * return a result of different type than the entity that is being queried.
+     * This method allows the subsequent {@link PageRequest} to be returned for the
+     * type of entity class that is being queried.</p>
+     *
+     * @param <E>         entity class of the attributes that are used as sort criteria.
+     * @param entityClass entity class of the attributes that are used as sort criteria.
+     * @return a request for the next page.
+     */
+    <E> PageRequest<E> nextPageRequest(Class<E> entityClass);
+
+    /**
+     * <p>Returns a request for the {@link PageRequest#previous() previous} page,
+     * or <code>null</code> if it is known that there is no previous page.</p>
+     *
+     * @return a request for the previous page.
+     */
+    PageRequest<T> previousPageRequest();
+
+    /**
+     * <p>Returns a request for the {@linkplain PageRequest#previous() previous}
+     * page, or <code>null</code> if it is known that there is no previous page.</p>
+     *
+     * <p>This method is useful when {@linkplain Query query language} is used to
+     * return a result of different type than the entity that is being queried.
+     * This method allows the subsequent {@link PageRequest} to be returned for the
+     * type of entity class that is being queried.</p>
+     *
+     * @param <E>         entity class of the attributes that are used as sort criteria.
+     * @param entityClass entity class of the attributes that are used as sort criteria.
+     * @return a request for the previous page.
```

Comment:
```suggestion
     * <p>Returns a request for the {@linkplain PageRequest#previous() previous}
     * page, if {@link #hasPrevious()} indicates there might be a previous page.</p>
     *
     * <p>This method is useful when {@linkplain Query query language} is used to
     * return a result of different type than the entity that is being queried.
     * This method allows the subsequent {@link PageRequest} to be returned for the
     * type of entity class that is being queried.</p>
     *
     * @param <E>         entity class of the attributes that are used as sort criteria.
     * @param entityClass entity class of the attributes that are used as sort criteria.
     * @return a request for the previous page.
     * @throws NoSuchElementException if it is known that there is no previous page.
     *         To avoid this exception, check for a {@code true} result of
     *         {@link #hasPrevious()} before invoking this method.
```